EduNLP/Pretrain/auto_vec.py

```diff
@@ -0,0 +1,303 @@
+import os
+from typing import List, Union
+from transformers import AutoModelForMaskedLM
+from transformers import DataCollatorForLanguageModeling, DataCollatorWithPadding
+from transformers import Trainer, TrainingArguments
+from copy import deepcopy
+
+from ..ModelZoo.hf_model import (
+    HfModelForPropertyPrediction,
+    HfModelForKnowledgePrediction,
+)
+from .pretrian_utils import EduDataset
+from .hugginface_utils import TokenizerForHuggingface
+
+__all__ = [
+    "EduAutoTokenizer",
+    "EduAutoDataset",
+    "finetune_edu_auto_model",
+    "finetune_edu_auto_model_for_property_prediction",
+    "finetune_edu_auto_model_for_knowledge_prediction",
+]
+
+DEFAULT_TRAIN_PARAMS = {
+    # default
+    "output_dir": None,
+    "num_train_epochs": 1,
+    "per_device_train_batch_size": 32,
+    # "per_device_eval_batch_size": 32,
+    # evaluation_strategy: "steps",
+    # eval_steps:200,
+    "save_steps": 1000,
+    "save_total_limit": 2,
+    # "load_best_model_at_end": True,
+    # metric_for_best_model: "loss",
+    # greater_is_better: False,
+    "logging_dir": None,
+    "logging_steps": 5,
+    "gradient_accumulation_steps": 1,
+    "learning_rate": 5e-5,
+    # disable_tqdm: True,
+    # no_cuda: True,
+}
+
+
+class EduAutoTokenizer(TokenizerForHuggingface):
+    """
+    Examples
+    ----------
+    >>> tokenizer = EduAutoTokenizer(add_special_tokens=True)
+    >>> item = "有公式$\\FormFigureID{wrong1?}$，如图$\\FigureID{088f15ea-xxx}$,\
+    ... 若$x,y$满足约束条件公式$\\FormFigureBase64{wrong2?}$,$\\SIFSep$，则$z=x+7 y$的最大值为$\\SIFBlank$"
+    >>> token_item = tokenizer(item)
+    >>> print(token_item.input_ids)
+    tensor([[ 101, 1062, 2466, 1963, 1745,  138,  100,  140,  166,  117,  167, 5276,
+             3338, 3340,  816, 1062, 2466,  102,  168,  134,  166,  116,  128,  167,
+             3297, 1920,  966,  138,  100,  140,  102]])
+    >>> print(tokenizer.tokenize(item)[:10])
+    ['公', '式', '如', '图', '[', '[UNK]', ']', 'x', ',', 'y']
+    >>> items = [item, item]
+    >>> token_items = tokenizer(items, return_tensors='pt')
+    >>> print(token_items.input_ids.shape)
+    torch.Size([2, 31])
+    >>> print(len(tokenizer.tokenize(items)))
+    2
+    >>> tokenizer.save_pretrained('test_dir') # doctest: +SKIP
+    >>> tokenizer = EduAutoTokenizer.from_pretrained('test_dir') # doctest: +SKIP
+    """
+
+    pass
+
+
+class EduAutoDataset(EduDataset):
+    pass
+
+
+def finetune_edu_auto_model(

```

Comment:
Should it be sth like `pretrain_hf_auto_model`? It is only used for huggingface models. Also, it is domain pretraining instead of fine-tuning?

Reply:
And the file name should be hf_auto_vec? It is not auto for our educational models

Reply:
And pretrain_bert, finetune_bert_for_xx should be the same with these auto functions? Maybe with these auto function, we can delete the bert_vec? Not sure if it is better or not. Or we can keep that file but directly reuse these auto functions?